### ISSUE TITLE
Security recommendation/hardening guide for applications that use Kubernetes client

### ISSUE THREAD

**Issue by AnshumanTripathi** [kubernetes/sig-security#121]
Create a security recommendation/hardening guide for applications that use the Kubernetes client. This could include different use-cases like:

1. Creating an application which runs `kubectl` commands.
2. An application that [uses the Kubernetes client](https://github.com/kubernetes/client-go)
3. A Kubernetes Operator

**Comment by chadmcrowell** [comment]
Hello, I'd like to contribute here. Just to be clear, the guide would focus on the security recommendations for applications interacting with the Kubernetes API? I can contribute in the following ways to recommend:
- handling of sensitive data (e.g. config, secrets, user creds)
- corruption of data or tampering
- security controls for principle of least privilege
- tightening up RBAC
- dependency control and vulnerability management
- enforcing network policy
- security of apps to maximize availability
- reduce attack vectors, and maintaining isolated workloads
- compliance and audit requirements
- supply chain attack prevention
- establish zero-trust for apps interacting with Kubernetes

**Comment by vaibhav2107** [comment]
Is this issue resolved by PR, if yes then we can close this issue

**Comment by tabbysable** [comment]
@chadmcrowell Yes, that sounds like exactly the sort of thing we were thinking of!

In general, there is a lot more security guidance for cluster administrators, but quite a bit less for folks who are writing an application to interact with their kubernetes cluster.
